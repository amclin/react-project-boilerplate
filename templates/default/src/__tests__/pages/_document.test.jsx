import React from 'react'
import * as Document from 'next/document'
// import { render } from '@testing-library/react'
import MyDocument from '../../pages/_document'

jest.mock('next/document')

// eslint-disable-next-line
Document.Head.mock = jest.fn(({ children }) => <head>{children}</head>)

describe('_document', () => {
  it('runs without error', () => {
    // tests should be improved when someone can spend time discovering how
    // const { asFragment } = render(<MyDocument />)
    // expect(asFragment()).toMatchSnapshot()
    const obj = new MyDocument()
    expect(obj.render()).toMatchSnapshot()
  })
})
