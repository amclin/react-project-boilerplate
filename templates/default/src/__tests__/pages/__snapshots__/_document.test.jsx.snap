// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_document runs without error 1`] = `
<Html
  lang="en"
>
  <Head>
    <Favicon />
    <Manifest />
  </Head>
  <body>
    <Main />
    <NextScript />
  </body>
</Html>
`;
